api/routes/generate.js

```diff
@@ -10,35 +10,30 @@ var thunkify = require('thunkify');
 // CRX packaging module, instantiated with the `new` keyword
 var Extension = require('crx');
 
+var knownHosts = {
```

Comment:
I don't agree -- when it's time to override the icon, title, or both for a new hostname, we now have to add 3 things:

1. An element to the `knownHosts` object
2. A case to `getExtensionTitle` with the element name and the title we want to override with
3. A case to `getExtensionIcon` with the element name and the icon we want to override with

Instead of just another `case` in the original `switch-case` statement with the hostname, title and icon, grouped together in 3 lines and easier to maintain.

In addition, we now have 2 switch-cases for the same object, `parsedUrl.hostname`, which is "duplicate-code".